doc/netplan-yaml.md

```diff
@@ -374,11 +374,19 @@ Match devices by MAC when setting options like: `wakeonlan` or `*-offload`.
   > (networkd backend only) Allow the specified interface to be configured even
   > if it has no carrier.
 
-- **critical** (bool)
+- **critical** (scalar)
 
-  > Designate the connection as "critical to the system", meaning that special
-  > care will be taken by to not release the assigned IP when the daemon is
-  > restarted. (not recognized by NetworkManager)
+  > This option sets the systemd KeepConfiguration option.
+  > (not recognized by NetworkManager)
+  >
+  > When set to "static", static addresses and routes won't be dropped on
+  > starting up process.
+  > When set to "dhcp-on-stop", addresses and routes won't be dropped when
+  > stopping the daemon.
+  > When set to "dhcp", addresses and routes provided by a DHCP server will
+  > never be dropped even if the DHCP lease expires, implies "dhcp-on-stop".
+  > When set to "yes", "dhcp" and "static" is implied.
```

Comment:
question: Did you double-check if something similar to `KeepConfiguration=` exists for NetworkManager these days? I'd like to align the wording between the two backends. If it does not exist it might be fine re-using the systemd-networkd terminology here. I need to think about this a bit more.